src/layers/base-layer.ts

```diff
@@ -73,7 +74,8 @@ export type VisualChannelField = Field | null;
 export type VisualChannelScale = keyof typeof SCALE_TYPES;
 
 export type LayerBaseConfig = {
-  dataId: string | null;
+  // TODO: Decide can dataId be null
+  dataId: string;
```

Comment:
`dataId` can be null

Reply:
I propose to bring this edit into a separate PR, since in many lines of code this field is used as an index when accessing an object